### DISCUSSION TITLE
Kamstrup Data Decode

### DISCUSSION THREAD

**Discussion by KeenanGeorge03** [wmbusmeters/wmbusmeters#600] [Q&A]
Hi 

Im trying to decode the following kamstrup meter data:

./wmbusmeters --format=json 22442d2c646892251b168d208ba4366820664e9879ccd3b4b533342dad5777ab3965B5 auto m
eter multical21 25926864 4B19F1C4122A384DBD7C99A753CEA64F

However im getting a CRC error & the meter KEY is correct.

Please point me in the right direction

Regards

**Comment by weetmuts** [comment]
You are quite right. Which version of wmbusmeters do you use and which dongle do you use to receive the telegram.
Is it T1 or C1 link mode?

I just fixed a bug in rc1180 where the rssi was added to the end of the telegram. Which might be what happens in your case. If you change the first 22 to 21 and remove the last B5 it decodes fine.

Can you test the latest github version?

**Reply by KeenanGeorge03** [reply]
Hi thanks ive pull the latest docker image & it converted successfully by manipulating the data as suggested.

Note we are not using a dongle but a data-concentrator that use mode C1.

Does this mean we will have to manipulate the mbus payload?

**Comment by weetmuts** [comment]
What is the setup? Do you invoke wmbusmeters from your own code, or do you have wmbusmeters query the mbus itself?

**Reply by KeenanGeorge03** [reply]
Hi the concentrator wakes up & listen for incoming messages & send it up as is to the server where we use your service to decode the data.

**Comment by weetmuts** [comment]
It seems like you have to ask the person/company responsible for this concentrator about the extra byte at the end.
But if they cannot fix it, I suppose it is up to you to remove the extra byte before feeding into wmbusmeters.

**Reply by KeenanGeorge03** [reply]
Hi thanks the above has been resolved. Another question why are we getting unknow driver for telegram below 4E4424231294766524067AB6004025AC47207E4D7329C719FA0B5435A283DB035E0934F7E1E8A
7785D88D538907010CE30C9632DA97F86B185895F1556CFB17ABACECFD917720A9F97A2C43A502C49
